### ISSUE TITLE
breqn is incompatible with unicode-math

### ISSUE THREAD

**Issue by intractabilis** [wspr/breqn#44]
Consider the following document:
```
\documentclass[english, 12pt, letterpaper]{scrartcl}

\usepackage{breqn}
\usepackage{unicode-math}

\begin{document}
\begin{dmath}
    T \left(h\right)
\end{dmath}

\end{document}
```

It results in "Missing \endgroup inserted" error. If you look at [this](https://www.latex-project.org/publications/2009-MHo-TUG-breqn-next-steps.pdf) 2009 document, it says: "Unicode is difficult, but don't worry, it’s already been done in the development version".

So, is this a bug or a feature?

**Comment by intractabilis** [comment]
Is there any hope?

**Comment by cvanelteren** [comment]
Bump?

**Comment by EmmanuelCharpentier** [comment]
I got bitten by the very same problem. Using `unicode-math` and `breqn` entails an error at the first use of the `\[` control sequence, 

- Lualatex : cryptic message `<recently read > \  Undefined control sequence`. This does *not* depend of the order of declaration.
- XeTeX : No errort, but no line btreak either.

Minimal (non-)working example : 

```
\documentclass[12pt]{article}

\usepackage{breqn}
% Loads fontspec and unicode math, sets NewCM fonts as defaults.
\usepackage[default]{fontsetup} 
% \usepackage[T1]{fontenc}
% \usepackage[utf8]{inputenc}

\begin{document}

Tagada

% \[ X\,=\,a+b+c+d+e+f+g+h+i+j+k+l+m+n+o+p+q+r+s+t+u+v+w+x+y+z+
%   a_0+b_0+c_0+d_0+e_0+f_0+g_0+h_0+i_0+j_0+k_0+l_0+m_0+n_0+
%   o_0+p_0+q_0+r_0+s_0+t_0+u_0+v_0+w_0+x_0+y_0+z_0 \]

\begin{dmath}[label={eq:1}]
  X\,=\,a+b+c+d+e+f+g+h+i+j+k+l+m+n+o+p+q+r+s+t+u+v+w+x+y+z+
  a_0+b_0+c_0+d_0+e_0+f_0+g_0+h_0+i_0+j_0+k_0+l_0+m_0+n_0+
  o_0+p_0+q_0+r_0+s_0+t_0+u_0+v_0+w_0+x_0+y_0+z_0
\end{dmath}

Tsoin-tsoin !

\end{document}

% Local Variables:
% TeX-engine: luatex
% End:

```

**Comment by FrankMittelbach** [comment]
@davidcarlisle go ahead David, if you know what can without too much surgery to improve the situation.

**Comment by EmmanuelCharpentier** [comment]
As far as I understand you, fixing the srut setting : 
- will avoid the error at load time with LuaLaTeX, but
- will *not* provide `breqn` unctionality to XeLaTeX/LuaLaTeX users.

Fixing the whole thing should be better, or, possibly, proposing a replacement geared to Unicode users, if maintaining compatibility with 8-bit (pdf)LaTeX is too difficult.

**Comment by davidcarlisle** [comment]
@EmmanuelCharpentier  yes in particular the first checkin made so far doesn't fix the main report here, but hopefully it's a start. Small steps...
